### ISSUE TITLE
About 1-shot and 5-shot

### ISSUE THREAD

**Issue by yankun-pku** [leokarlin/LaSO#13]
Hi, could I ask where reflects the 1-shot and 5-shot setting in your codes?

**Comment by ZhangYikaii** [comment]
LaoGe, I had a similar problem: #14

**Comment by alfassy** [comment]
@yankun-pku You will have to be more specific as this is being reflected in so many different parts of the code. we have training and few different types of tests

**Comment by yankun-pku** [comment]
> @yankun-pku You will have to be more specific as this is being reflected in so many different parts of the code. we have training and few different types of tests

I don't think so. First, there is no code of 5-shot setting. Second, for 1-shot setting, the training procedure is just in a traditional image classification  style, where the restriction is only one image one time.

**Comment by ZhangYikaii** [comment]
Sorry for interrupting, "for 1-shot setting, the training procedure is just in a traditional image classification style" I guess the author uses a similar Few-Shot sampling method [here](https://github.com/leokarlin/LaSO/blob/master/scripts_coco/test_augmentation.py#L211).

But I'm **still trying** to understand the difference between 1-shot and 5-shot. I mean [here](https://github.com/leokarlin/LaSO/blob/master/oneshot/alfassy/datasets.py#L695) doesn't seem to work for different sampling method.

**Comment by alfassy** [comment]
from the readme: "Generate the augmentation model's results
Paper model 1 shot -

$ python test_augmentation.py --batch_size=4 --class_cap=1 --class_ind_dict_path='<LaSO folder>/data_for_augmentation/1shotRun1ClassIdxDict.pkl' --crop_size=299 --latent_dim=2048 --lr=0.01 --n_epochs=50 --resume_path=<path_to_LaSO_models> --used_ind_path='<LaSO folder>/data_for_augmentation/1shotRun1UsedIndices.pkl' --results_path=<folder path to save models> --coco_path=<path to local coco folder>
Paper model 5 shot -

$ python test_augmentation.py --batch_size=4 --class_cap=5 --class_ind_dict_path='<LaSO folder>/data_for_augmentation/5shotRun1ClassIdxDict.pkl' --crop_size=299 --latent_dim=2048 --lr=0.01 --n_epochs=50 --resume_path=<path_to_LaSO_models> --used_ind_path='<LaSO folder>/data_for_augmentation/5shotRun1UsedIndices.pkl' --results_path=<folder path to save models> --coco_path=<path to local coco folder>
"

**Comment by alfassy** [comment]
The 3 differences between 1 and 5 shot in the augmentation scenario is: a. class_cap, b. class_ind_dict_path, c.used_ind_path.
open the files of b and c in debug mode, and you'll understand, c holds the indices per class in each 1/5 shot scenario, so you'll have more indices in the 5 shot scenario, it's called usedIndices in the dataset class.

**Comment by ZhangYikaii** [comment]
@alfassy Great reply. I got it. Thank you very much!!
